crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -455,6 +459,39 @@ pub(crate) mod tests {
         RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap()
     }
 
+    fn insert_room_rule(
+        rules: &mut Rules,
+        kind: RuleKind,
+        room_id: &RoomId,
+        notify: bool,
+    ) -> Result<Option<Command>, NotificationSettingsError> {
+        let command = rules.build_insert_rule_command(kind, room_id, notify)?;
+        if let Some(inner) = &command {
+            rules.apply(inner);
+        }
+        Ok(command)
+    }
```

Comment:
With the suggestion above, this could be replaced with something like `rules.apply(Commands::new().insert_rule(kind, room_id, notify)?)`, and we wouldn't need the test helpers anymore.